src/Neo/Wallets/NEP6/NEP6Wallet.cs

```diff
@@ -293,12 +293,18 @@ public override WalletAccount Import(string nep2, string passphrase, int N = 163
         /// </summary>
         public JObject ToJson()
         {
+            NEP6Account[] accountValues;
+            lock (accounts)
```

Comment:
I prefer locking to be on its own object altogether. The reason for this is so that way it frees it up quicker and it doesn't Lock up when something wants to read it. Only prevents writing when saving the Json

Reply:
Lock time wasted procesing .ToJson

Reply:
This is not thread safe if `ToJson` and `ChangePassword`  concurrently, because `account.ToJson` read fields of the `Account` and `ChangePassword` changes fields of the `Account`.

Reply:
> Lock time wasted procesing .ToJson

This is not performance critical path, so excessive optimization is not needed.